tests/system/model_monitoring/test_model_monitoring.py

```diff
@@ -0,0 +1,404 @@
+import json
+import os
+import string
+from datetime import datetime, timedelta
+from random import choice, randint, uniform
+from time import monotonic, sleep
+from typing import Optional
+
+import pandas as pd
+import pytest
+from sklearn.datasets import load_iris
+from v3io.dataplane import RaiseForStatus
+from v3io_frames import frames_pb2 as fpb2
+from v3io_frames.errors import CreateError
+
+import mlrun
+import mlrun.api.crud
+import mlrun.api.schemas
+from mlrun.api.schemas import (
+    ModelEndpoint,
+    ModelEndpointMetadata,
+    ModelEndpointSpec,
+    ModelEndpointStatus,
+)
+from mlrun.config import config
+from mlrun.errors import MLRunNotFoundError
+from mlrun.utils.model_monitoring import parse_model_endpoint_store_prefix
+from mlrun.utils.v3io_clients import get_frames_client, get_v3io_client
+from tests.system.base import TestMLRunSystem
+
+
+# Marked as enterprise because of v3io mount and pipelines
+@TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.enterprise
+class TestModelMonitoringAPI(TestMLRunSystem):
+    def test_clear_endpoint(self):
+        endpoint = self._mock_random_endpoint()
+        db = mlrun.get_run_db()
+
+        db.create_or_patch_model_endpoint(
+            endpoint.metadata.project, endpoint.metadata.uid, endpoint
+        )
+
+        endpoint_rsp = db.get_model_endpoint(
+            endpoint.metadata.project, endpoint.metadata.uid
+        )
+        assert endpoint_rsp
+        assert endpoint_rsp.metadata.uid == endpoint.metadata.uid
+
+        db.delete_model_endpoint_record(
+            endpoint.metadata.project, endpoint.metadata.uid
+        )
+
+        # test for existence with "underlying layers" functions
+        with pytest.raises(MLRunNotFoundError):
+            endpoint = mlrun.api.crud.ModelEndpoints().get_endpoint(
+                auth_info=self._get_auth_info(),
+                project=endpoint.metadata.project,
+                endpoint_id=endpoint.metadata.uid,
+            )
+
+    def test_store_endpoint_update_existing(self):
+        endpoint = self._mock_random_endpoint()
+        db = mlrun.get_run_db()
+
+        db.create_or_patch_model_endpoint(
+            endpoint.metadata.project, endpoint.metadata.uid, endpoint
+        )
+
+        endpoint_before_update = db.get_model_endpoint(
+            endpoint.metadata.project, endpoint.metadata.uid
+        )
+
+        assert endpoint_before_update.status.state is None
+
+        endpoint_before_update.status.state = "testing...testing...1 2 1 2"
```

Comment:
```suggestion
        updated_state = "testing...testing...1 2 1 2"
        endpoint_before_update.status.state = updated_state
```
(and also use later in the validation)